src/components/ListV2/ListContext.tsx

```diff
@@ -78,14 +73,27 @@ export const ListProvider = <T,>({
     [isSelectable, template],
   )
 
+  const onChangeSelectedIds: Dispatch<SetStateAction<string[]>> = useCallback(
+    (value: string[] | ((currentIds: string[]) => string[])) => {
+      if (onSelectedIdsChangeRef.current) {
+        onSelectedIdsChangeRef.current(
+          typeof value !== 'function' ? value : value(selectedIds),
+        )
+      } else {
+        setSelectedIds(typeof value !== 'function' ? value : value(selectedIds))
+      }
```

Comment:
To avoid duplication:
```suggestion
      const newSelectedIds = typeof value !== 'function' ? value : value(selectedIds)
      
      if (onSelectedIdsChangeRef.current) {
        onSelectedIdsChangeRef.current(newSelectedIds)
      } else {
        setSelectedIds(newSelectedIds)
      }
```